### PR TITLE
Upgrade ingress api version to networking.k8s.io/v1

### LINKED ISSUE #4068: Upgrade Ingress API version to networking.k8s.io/v1
**What's it about?**
Upgrade Ingress API to networking.k8s.io/v1.  

**What's the reason why we need it?**
Ingress is the core component that is used by the KubeSphere Routes feature. KubeSphere is still using the old api "extensions/v1beta1" both Frontend and Backend. However "extensions/v1beta1" Ingress  is deprecated in v1.14+, unavailable in v1.22+. So we need to upgrade it to the latest version to be compatible with the higher version Kubernetes cluster.

**Area Suggestion**
/area console
/area microservice
/area multicluster

/kind feature-request

/good-first-issue

### PR DESCRIPTION
**Pull request by wenhuwang**
Signed-off-by: wenhuwang <976400757@qq.com>

### What type of PR is this?
/kind api-change
/kind feature

### What this PR does / why we need it:
Ingress is the core component that is used by the KubeSphere Routes feature. KubeSphere is still using the old api "extensions/v1beta1" both Frontend and Backend. However "extensions/v1beta1" Ingress is deprecated in v1.14+, unavailable in v1.22+. So we need to upgrade it to the latest version to be compatible with the higher version Kubernetes cluster.

### Which issue(s) this PR fixes:

Fixes #4068

### Special notes for reviewers:
```
```

### Does this PR introduced a user-facing change?

```release-note
None
```

### Additional documentation, usage docs, etc.:

```docs

```

### CODE REVIEW

File: staging/src/kubesphere.io/api/types/v1beta1/federatedingress_types.go [hunk 1]
```diff
@@ -45,7 +45,7 @@ type FederatedIngressSpec struct {
 }
 
 type IngressTemplate struct {
-	Spec extensionsv1beta1.IngressSpec `json:"spec,omitempty"`
+	Spec networkingv1.IngressSpec `json:"spec,omitempty"`
```
Comment (RolandMa1986) [inline staging/src/kubesphere.io/api/types/v1beta1/federatedingress_types.go:48]: @yuswift @wenhuwang 
Needs to consider FederatedIngress API version conversion. Build-in Ingress resources could be handled with Kube-apiserver by default. But not for CRD.

Comment (yuswift) [reply staging/src/kubesphere.io/api/types/v1beta1/federatedingress_types.go:48]: We don't use the file to generate crd files. So I think we can skip it.

Comment (wenhuwang) [reply staging/src/kubesphere.io/api/types/v1beta1/federatedingress_types.go:48]: Subsequent ingress will use the network.k8s.io/v1 version, I think IngressTemplate should also be upgraded together

Comment (yuswift) [reply staging/src/kubesphere.io/api/types/v1beta1/federatedingress_types.go:48]: Yes, it should be upgraded.

**Pull request merged**